byterun/caml/roots.h

```diff
@@ -35,10 +35,48 @@ CAMLextern void caml_do_local_roots (scanning_action, value *, value *,
 CAMLextern void caml_do_local_roots(scanning_action f, char * bottom_of_stack,
                                     uintnat last_retaddr, value * gc_regs,
                                     struct caml__roots_block * local_roots);
-#endif
+#endif /* NATIVE_CODE */
 
 CAMLextern void (*caml_scan_roots_hook) (scanning_action);
 
+
+/* Hooks to profile systhreads. */
+
+#define ST_NEXT_HOOK(thread_id,bottom,retaddr) \
```

Comment:
The name of the macro (and the hook) is really weird. Since it's about scanning roots, it should be called something like `ST_ROOT_SCAN_HOOK` or something.